ClientCore/Extensions/StringExtensions.cs

```diff
@@ -77,4 +77,42 @@ public static string L10N(this string defaultValue, string key, bool notify = tr
         => string.IsNullOrEmpty(defaultValue)
             ? defaultValue
             : Translation.Instance.LookUp(key, defaultValue, notify);
+
+    /// <summary>
+    /// Replace special characters with spaces in the filename to avoid conflicts with WIN32API.
+    /// </summary>
+    /// <param name="defaultValue">The default string value.</param>
+    /// <returns>File name without special characters or reserved combinations.</returns>
+    /// <remarks>
+    /// Reference: <a href="https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file">Naming Files, Paths, and Namespaces</a>.
+    /// </remarks>
+    public static string ToWin32FileName(this string defaultValue)
+    {
+        foreach (char ch in "/\\:*?<>|")
+            defaultValue = defaultValue.Replace(ch, ' ');
+
+        // If the user is somehow using "con" or any other filename that is
+        // reserved by WIN32API, it would be better to rename it.
+
+        string[] reservedFileNames = 
+        {
+            "CON",
+            "PRN",
+            "AUX",
+            "NUL",
+            "COM1", "COM2", "COM3", "COM4", "COM5", "COM6", "COM7", "COM8", "COM9", "COM¹", "COM²", "COM³",
+            "LPT1", "LPT2", "LPT3", "LPT4", "LPT5", "LPT6", "LPT7", "LPT8", "LPT9", "LPT¹", "LPT²", "LPT³"
+        };
+
+        foreach(var reserved in reservedFileNames)
```

Comment:
there should be a space between foreach and left bracket

also I think it is better to use a HashSet (public static readonly IReadOnlySet) or an ImmutableList (public static readonly ImmutableList) instead of a list here

Reply:
> also I think it is better to use a HashSet (public static readonly IReadOnlySet) or an ImmutableList (public static readonly ImmutableList) instead of a list here

Ok, where should i place this hashset?

Reply:
> > also I think it is better to use a HashSet (public static readonly IReadOnlySet) or an ImmutableList (public static readonly ImmutableList) instead of a list here
> 
> Ok, where should i place this hashset?

Sorry, I just realized that IReadOnlySet is not support in .NET 4.8. Therefore a static public class member is not a good idea anymore.

Speaking out of the current project, for a .net 8 project, we can write something like this:

```cs

public static class StringExtensions
{
    public static readonly IReadOnlySet<string> InvalidWin32FileNames = new HashSet<string>();
}
```